### ISSUE TITLE
403 for viewing the source, access denied for installing.

### ISSUE THREAD

**Issue by sapioit** [OpenUserJS/OpenUserJS.org#1861]
**What happened?**

  I click on a/any script which isn't my own. I click on `Source Code`, then I see the first screenshot provided. If I click on `Install` instead, I see the second screenshot provided.

**What did you expect to happen?**

  Seeing the source, like for my own scripts but without the option to save the changes, or even the option to change the contents of the text-area (using `read-only` in the HTML), for the `Source Code` button, and seeing the raw file (like here on Github) for the `Install` option.

**What the problem might be.**

  Either I am banned by the server and unable to see the sources because of that (if the source is loaded client-side in the `Source Code` page), or you have problems with the servers where you hold the sources or with the connection between the sources and the website.

**Screenshots**

![screenshot 12767](https://user-images.githubusercontent.com/3173159/143502877-65791a21-5ab5-430e-b596-70334958469e.png)
![screenshot 12768](https://user-images.githubusercontent.com/3173159/143502885-1b3edf09-a14a-43c2-acea-6fa25fea81dd.png)

**Comment by sapioit** [comment]
Update: I think it has to do with the links themselves, too. For example, I can get a few lines of code in the first 2 links (which isn't on the website), but not not on the next 3 links (the first of which is on the website). So the `.meta.js` works, but the `.install.js`, the `.user.js`, and the `.install.js` don't work.

https://openuserjs.org/meta/DecentM/No_YouTube_notifications.meta.js
https://openuserjs.org/install/DecentM/No_YouTube_notifications.meta.js
https://openuserjs.org/install/DecentM/No_YouTube_notifications.user.js
https://openuserjs.org/install/DecentM/No_YouTube_notifications.user.js#bypass=true
https://openuserjs.org/install/DecentM/No_YouTube_notifications.install.js

**Comment by Martii** [comment]
> I click on a/any script which isn't my own.

Not true.

> I click on `Source Code`, then I see the first screenshot provided.

Usually that happens on other Authors pages that don't have `@updateURL` formatted properly. We are in lockdown as denoted at https://openuserjs.org/about . You also have your Author Tools notification that has that enhanced and a link to where it's documented. But since you didn't go there ... https://openuserjs.org/about/Frequently-Asked-Questions#q-does-openuserjs-org-have-meta- And if you are the Author you should read the text under the 403 to fix it while we are in this state. This is nothing new and has been available for years. If you are not the Author then you get to wait on them.

> Either I am banned by the server and unable to see the sources because of that (if the source is loaded client-side in the Source Code page),  ...

You *(everyone is)* are not banned just limited atm.


> or you have problems with the servers where you hold the sources or with the connection between the sources and the website.

Interesting guess however not the reality. Read the links that I duplicated here for you.

I am still tweaking this routine a bit so there might be some bumps but it is a holiday weekend here so might be a bit.

**Comment by sapioit** [comment]
> Usually that happens on other Authors pages that don't have `@updateURL` formatted properly. We are in lockdown as denoted at https://openuserjs.org/about . You also have your Author Tools notification that has that enhanced and a link to where it's documented. But since you didn't go there ... https://openuserjs.org/about/Frequently-Asked-Questions#q-does-openuserjs-org-have-meta- And if you are the Author you should read the text under the 403 to fix it while we are in this state. This is nothing new and has been available for years. If you are not the Author then you get to wait on them.

**TL;DR: As I mentioned below, you could make it so every single script which doesn't have an `@updateURL` gets one inserted right below `// ==UserScript==`, using the link we open by clicking `Install`, and not put the burden of updating old scripts on forgotten accounts, dead people, and authors who have given up on maintaining their older scripts. The current behaviour of the feature makes it look like a bug. Because features can be bugs, too.**

I did see the message, but I still think that being completely unable to use that script is not user-friendly. At most, you could automatically generate an `@updateURL` based on the file on your website, and add it to all the userscripts which don't have it. There should also be a warning that the script doesn't have an update URL but still show us the source, even if we have to consent to seeing the source of a script without an `@updateURL`. 

Not only that, but a few weeks ago, it worked perfectly even if the scripts didn't have an `@updateURL`. I did read that page, but I still think that, by design, the "feature" acts like a bug.

I'm not trying to be condescending, nor am I angry, I am just sad that the website decided to give up the support for most of it's older userscripts by expecting long-forgotten scripts and accounts to suddenly become active again, especially since some of them might sadly not even exist anymore (people die from old age, accidents, and illnesses, and perfectly-healthy living people forget the passwords for their email addresses, or even the email addresses themselves, too).

**Comment by Martii** [comment]
> **TL;DR:...**

TL;AR: Definitely agree there on the too long part and :-1: for the suggestion.

Your suggestion is something OUJS doesn't do and has already been talked about. This is how it is done but at least you've had your soap box moment. :smile_cat:
